### DISCUSSION TITLE
Cleaning up Flow Forecast

### DISCUSSION THREAD

**Discussion by isaacmg** [AIStream-Peelout/flow-forecast#804] [Announcements]
## Cleaning up Flow-Forecast
It's crazy to say but it has been now five+ years since I started coding Flow Forecast. Overtime the repository has evolved and its focus has expanded. Looking back in retrospect there are a lot of things that both myself and other contributors could probably have implemented better. There are also a lot of things that have over-time just gotten sloppy. The folder structure makes no sense, parameter names are not standardized in the initialization, documentation is lacking, and there is an over emphasis on notebooks versus running as a CLI tool. As an open source repository we also have too many issues with no details and old issues. While, we will continue to expand and add new time series models (including foundation and multi-models) I plan on devoting significant time to "code hygiene" and updating our contributing guidelines. I definitely primarily guilty of t

## Contributing Guidelines
1. Going forward all new code MUST have type hints and docstrings. The docstrings must be in Sphinx compatible format. For new models and loss functions the docstring should include information a link to the original paper. Additionally, type hints on PyTorch tensors must use jaxtyping and specify what each dimension of the tensor is. For instance (batch_size, forecast_history, etc).
2. Hard to understand code should be commented. We don't need to comment every line or operation but particularly complex things like tensor manipulation where dimensions are changed should be commented.
3. All new pull requests with the exception of automatic updates must have a linked issue or issues. The PR must also detail
